### PR TITLE
Cleanup and refactoring throughout the Baja codebase

### PR DESCRIPTION
**Pull request by dominic-mulligan-arm**
1. Added inline Rustdoc comments everywhere, throughout the library.
2. Factored out important constants (e.g. the cryptographic certificate template into explicit constants declared in the source file.
3. Generalized the `VeracruzIdentity` type in `veracruz-utils` and used this type in `Baja` in place of an explicit triple of `u32`, `Certificate`, and `Vec<VeracruzRole>` values.
4. Removed uneccessary allocations/clones throughout the codebase, and other bits of Rust cleanup preferring reference passing rather than value passing where we can get away with it.
5. Added `inline` hints for small functions.
6. Removed unnecessary error type on the creation of Baja sessions, and propagated this change onwards.
7. (Trivial) removed non-breaking space characters in various RustDoc comments (across the Veracruz codebase) which cause RustDoc to not render correctly in favour of a proper space character.
8. Generalized some functions taking Strings as inputs to functions that take generic arguments that can be converted into Strings.

Various other small improvements throughout.

### CODE REVIEW

File: baja/src/baja_session.rs
Comment (ShaleXIONG) [inline baja/src/baja_session.rs:31]: This `Principal` type is duplicated with `VeracruzIdentity` in veracruz_util. One trick that might work is `type Principal = VeracruzIdentity` but I am not sure if we can add more methods to the `Principal` type.

Comment (dominic-mulligan-arm) [reply baja/src/baja_session.rs:31]: Good point.  I'll see whether there's some unification that can be done, here.

Comment (dominic-mulligan-arm) [reply baja/src/baja_session.rs]: I managed to unify the two types by introducing a type-variable to abstract over the different representations of cryptographic certificates that the two used, i.e. either `String` or the `Certificate` type from the `RusTLS` library, and moved extra material across into `veracruz-utils`.

File: veracruz-utils/src/policy.rs
Comment (dreemkiller) [inline veracruz-utils/src/policy.rs:148]: What's the advantage of adding the type abstraction here?

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: c09b9442c95ae4afe7bb63cac8a70d936918412c
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=2b353059-7f4f-4184-affb-c6bd8eb97f0e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 7903687ade3d6288f3c2e0986f27e464c6168081
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=4aeb3a00-afb6-49f8-af86-b2a86561efe3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 4b4035014e2dcdcf19fbe2f0200b3afc3d42124a
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=2a7c20f3-6628-4180-b6fc-9fb38bfcabfe%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: veracruz-utils/src/policy.rs
Comment (dominic-mulligan-arm) [reply veracruz-utils/src/policy.rs:148]: The combination of client ID, certificate, and roles is used in a few places in Veracruz (e.g. `Baja` and `veracruz-utils`).  However, depending on where the type is used, the actual representation of the certificate is different: in Baja the certificate is represented as the `Certificate` type from RusTLS, whereas in veracruz-utils it is represented as an unparsed string that was read from the global policy.  By introducing the type-abstraction, we can use a single type for representing the client ID, certificate, roles combination whilst being ambivalent about the exact representation of the certificate.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: c5c5e9591d20cbff024b75e45248deb4379480df
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=665a136d-260f-4803-8e89-0456ef338d90%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: veracruz-utils/src/policy.rs
Comment (dreemkiller) [reply veracruz-utils/src/policy.rs:148]: Thanks, I see that now. I didn't see the other uses. Nice.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: b97a0f60ec2f83607a7b86b71ca9032354a06486
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=9e51221c-3681-47ba-bc41-d88687b54468%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 216309a85b315fbb4ea55e01d44548ac88f07ad7
* Result: SUCCEEDED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=50a6272c-a6cb-42ee-8c8d-d1419759b921%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: baja/src/baja.rs
Comment (ShaleXIONG) [inline baja/src/baja.rs:360]: I think the comment here should be in one continuous section as a suggested implementation strategy. Maybe just move all out of `{...}`and change the code to `let name = "ac40a0c".to_string();`.

**Review by ShaleXIONG — APPROVED** [review]

**Review by ShaleXIONG — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
Let's wait for @dreemkiller's comments/approval before merging on this.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: aa29ea8fb88a79105e04b151c1bb44c11a16c967
* Result: SUCCEEDED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=b4bd9d67-82c0-4f3e-8e75-c35f1476c00d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by dreemkiller — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
1+1=1, merging.

**Pull request merged**
